### PR TITLE
feat: replace iframe with chrome sidepanel

### PR DESCRIPTION
**Pull request by AdityaPimpalkar**
fixes - #5201

https://github.com/twentyhq/twenty/assets/13139771/871019c6-6456-46b4-95dd-07ffb33eb4fd

### CODE REVIEW

File: packages/twenty-chrome-extension/src/background/index.ts [hunk 1]
```diff
@@ -1,40 +1,55 @@
...
 import { exchangeAuthorizationCode } from '~/db/auth.db';
 import { isDefined } from '~/utils/isDefined';
 
 // Open options page programmatically in a new tab.
-chrome.runtime.onInstalled.addListener((details) => {
-  if (details.reason === 'install') {
-    openOptionsPage();
-  }
-});
+// chrome.runtime.onInstalled.addListener((details) => {
```
Comment (AdityaPimpalkar) [inline packages/twenty-chrome-extension/src/background/index.ts:7]: Commenting this for now, will replace it with a call to a dedicated page for chrome extension of twenty documentation

File: packages/twenty-chrome-extension/src/options/Options.tsx [hunk 2]
```diff
@@ -83,39 +105,37 @@ const Options = () => {
...
+            />
+            <MainButton
+              title="Connect your account"
+              onClick={() => authenticate()}
+              fullWidth
+            />
+            <MainButton
+              title="Sign up"
+              variant="secondary"
+              onClick={() => window.open(`${import.meta.env.VITE_FRONT_BASE_URL}`, '_blank')}
```
Comment (AdityaPimpalkar) [inline packages/twenty-chrome-extension/src/options/Options.tsx:134]: Fixed the signup link

**Comment by lucasbordeau** [comment]
@AdityaPimpalkar Could you look into that issue with @Weiko https://github.com/twentyhq/twenty/issues/5224 it's currently causing problem to validate this PR properly.

File: packages/twenty-chrome-extension/src/contentScript/extractPersonProfile.ts [hunk 3]
```diff
@@ -82,44 +83,60 @@ export const addPerson = async () => {
...
+    const personButtonSpan = personButtonDiv.getElementsByTagName('span')[0];
     const person = await checkIfPersonExists();
+
+    const openPersonOnSidePanel = async (personId: string) => {
+      await changeSidePanelUrl(
+        `${import.meta.env.VITE_FRONT_BASE_URL}/object/person/${personId}`,
+      );
+      personButtonSpan.textContent = 'View in Twenty';
+      personButtonDiv.onClickHandler(() => {
+        chrome.runtime.sendMessage({ action: 'openSidepanel' });
```
Comment (lucasbordeau) [inline packages/twenty-chrome-extension/src/contentScript/extractPersonProfile.ts:126]: Should change side panel url if already open ?

File: packages/twenty-chrome-extension/src/options/Options.tsx [hunk 4]
```diff
@@ -4,24 +4,31 @@ import styled from '@emotion/styled';
 import { Loader } from '@/ui/display/loader/components/Loader';
```
Comment (lucasbordeau) [inline packages/twenty-chrome-extension/src/options/Options.tsx:4]: Could we rename Options into a more self-explanatory naming ? Options seems very vague, SidePanel or something like that would be more appropriate.

**Review by lucasbordeau — COMMENTED** [review]
I can't go to the page in the side panel when it's already open, see linked video

https://github.com/twentyhq/twenty/assets/26528466/242130d2-5c25-4d59-af02-c58840c70371

+ We should disable selecting the text of the button : user-select: none;

File: packages/twenty-chrome-extension/src/options/Options.tsx (hunk 4 above)
Comment (AdityaPimpalkar) [reply packages/twenty-chrome-extension/src/options/Options.tsx]: Done 👍

**Review by FelixMalfait — APPROVED** [review]

**Pull request merged**